.github/workflows/cron-weekly-update-nightly.yml

```diff
@@ -0,0 +1,43 @@
+name: Update Nightly rustc
+on:
+  schedule:
+    - cron: "0 0 * * 0" # runs weekly on Sunday at 00:00
+  workflow_dispatch: # allows manual triggering
+jobs:
+  format:
+    name: Update nightly rustc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Update rust.yml to use latest nightly
+        run: |
+          set -x
+          # Not every night has a nightly, so extract the date from whatever
+          # version of the compiler dtolnay/rust-toolchain gives us.
```

Comment:
Is this putting a lot of trust in what comes downstream from dtolnay?

Reply:
It's the same amount of trust as in our existing CI scripts. (Which isn't much -- CI can't affect the repo.)